docs/details/signal.dox

```diff
@@ -211,32 +211,108 @@ respectively, then the possible batch operations are as follows.
 \defgroup signal_func_approx1 approx1
 \ingroup approx_mat
 
-approx1 interpolates data along the first dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
-
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n). The positions are sampled with respect to data at these
-locations.
+approx1 interpolates data along one dimension. Interpolation computes
+for values of unknown points out of known ones.
+
+The multi-dimensional (>=1D) input array, `in`, contains known values
+to be used for interpolation. These known values lie on a 1D
+interpolation grid - a one-dimensional, uniformly-spaced grid of
+points. Interpolation is performed to obtain values for any unknown
+points on the grid. The index values of the grid can be specified with
+the `grid_beg` and `grid_step` arguments.
+
+The position array (<=1D), `pos`, contains a combination of known and
+unkown positions/indices on the interpolation grid. The position
+array, `pos`, must be one-dimensional and must lead across the
+dimension in which the measurements will be made in the input array.
+
+Any unknown position/index values that lie within the interpolation
+grid range will be calculated for using interpolation. Unknown points
+that are outside of the interpolation grid range will be replaced with
+the `off_grid` value. Known points will be replaced with corresponding
+values from the input array.
+
+The interpolation dimension, `interp_dim`, is the leading dimension
+used to measure across the interpolation grid. Interpolation is thus
+performed along the same dimension of the input array.
+
+The uniformly-spaced one-dimensional interpolation grid on which the
+input values lie on can be adjusted with the corresponding `grid_beg`
+and `grid_step` arguments. A standard uniform interpolation grid with
+equally spaced indices in the range of [0,n) can be achieved with grid
+begin and step values of 0 and 1, respectively. A standard uniform
+grid will be used in the case that the interpolating dimension as well
+as the grid start and step values are not passed in.
```

Comment:
So I'm still not sure what `grid_beg` and `grid_step` are for, given that we have `pos` already. Is `grid_beg` where you start basing the interpolation? Say I have a 1D input array with 100 elements that look like a piecewise function, where the first half [0, 50) looks like a quadratic function and the second half [50, 100) looks like a line, and I set `pos` to be indices within the second half, and I don't want the first half to affect the interpolation - do I set `grid_beg` then to be 50?